packages/components/src/Button/IconButton/IconButton.tsx

```diff
@@ -0,0 +1,43 @@
+import React from "react"
+import {
+  GenericButton,
+  GenericProps,
+  ButtonFormAttributes,
+  WorkingProps,
+  WorkingUndefinedProps,
+  ButtonBadgeProps,
+} from "../GenericButton"
+
+export type IconButtonProps = GenericProps &
+  ButtonFormAttributes &
+  (WorkingProps | WorkingUndefinedProps) & {
+    label: string
+    primary?: boolean
+    destructive?: boolean
+    secondary?: boolean
+    /** @default "regular" */
+    size?: "small" | "regular"
+    badge?: ButtonBadgeProps
+    type?: "submit" | "reset" | "button"
+    fullWidth?: boolean
+    icon?: JSX.Element
+    disabled?: boolean
+  }
```

Comment:
Aren't most of these props on the generic props?

Almost seems like it should just be `export type IconButtonProps = Omit<GenericProps, "iconButton">`

Reply:
Types improvement is out of scope :D